frontend_v2/src/app/components/challenge/challengesettings/challengesettings.component.ts

```diff
@@ -212,6 +280,104 @@ export class ChallengesettingsComponent implements OnInit, OnDestroy {
+        SELF.leaderboardVisibility.icon = 'fa fa fa-toggle-off grey-text text-darken-1';
+       }
+      }
+  }
+
+  /**
+   * Leadeboard Visibility toggle function
+   */
+  toggleLeaderboardVisibility() {
+    const SELF = this;
+    let toggleLeaderboardVisibilityState, visibility;
+    if (SELF.leaderboardVisibility.state === 'Public') {
+      toggleLeaderboardVisibilityState = 'private';
+      visibility = 1;
+      SELF.leaderboardVisibility.state = 'Private';
+      SELF.leaderboardVisibility.icon = 'fa fa fa-toggle-off grey-text text-darken-1';
+    } else {
+      toggleLeaderboardVisibilityState = 'public';
+      visibility = 3;
+      SELF.leaderboardVisibility.state = 'Public';
+      SELF.leaderboardVisibility.icon = 'fa fa-toggle-on green-text';
+    }
+      const BODY: FormData = new FormData();
+      BODY.append("visibility", visibility);
+      SELF.apiService
+      .patchFileUrl(
+        SELF.endpointsService.particularChallengePhaseSplitUrl(SELF.selectedPhaseSplit['id']),
+        BODY
+      )
+        .subscribe(
+          (data) => {
+            SELF.selectedPhaseSplit['visibility'] = data.visibility;
+            SELF.challengeService.changePhaseSplitSelected(true);
+            if (visibility == 3) {
+              SELF.selectedPhaseSplit['showPrivate'] = false;
+              SELF.leaderboardVisibility.state = 'Public';
+              SELF.leaderboardVisibility.icon = 'fa fa-toggle-on green-text';
+            } else {
+              SELF.selectedPhaseSplit['showPrivate'] = true;
+              SELF.leaderboardVisibility.state = 'Private';
+              SELF.leaderboardVisibility.icon = 'fa fa-toggle-off grey-text text-darken-1';
+            }
+            SELF.globalService.showToast(
+              'success',
+              'The phase split was successfully made ' + toggleLeaderboardVisibilityState,
+              5
+            );
+          },
+          (err) => {
+            SELF.globalService.handleApiError(err, true);
+            SELF.globalService.showToast('error', err);
+            if (visibility == 3) {
+              SELF.leaderboardVisibility.state = 'Private';
+              SELF.leaderboardVisibility.icon = 'fa fa-toggle-off grey-text text-darken-1';
+            } else {
+              SELF.leaderboardVisibility.state = 'Public';
+              SELF.leaderboardVisibility.icon = 'fa fa-toggle-on green-text';
+            }
+          },
+          () => this.logger.info('LEADERBOARD-VISIBILITY-UPDATE-FINISHED')
+        );
+    }
+
+  /**
+   * For edit leaderboard schema
+   */
+  editLeaderboardSchema() {
+    
+    const SELF = this;
+    /**
+     * Parameters of the modal
+     */
+     const PARAMS = {
+      title: 'Edit Leaderboard Schema',
+      label: 'leaderboard_schema',
+      isEditorRequired: true,
+      confirm: 'Submit',
+      deny: 'Cancel',
+      confirmCallback: SELF.apiCall,
+    };
+    SELF.globalService.showModal(PARAMS);
+
+    }
```

Comment:
Let's remove this option for now. We can add this when I have the API ready. Probably next week, I won't have bandwidth to work on this before that

Reply:
Sure, I have removed it